### ISSUE TITLE
Provide mock possibility for `SecHubClient` in `sechub-api-java`

### ISSUE THREAD

**Issue by de-jcup** [mercedes-benz/sechub#2533]
## Situation
The development on the WebUI is currently always blocked by the development of the SecHub Java Api client (see https://github.com/mercedes-benz/sechub/tree/develop/sechub-api-java)

If the WebUI developers wants to use a new feature (or there is bug inside the Java API client) the Web development is always blocked at the moment - which can be very frustrating.

## Wanted
It shall be possible to develop the WebUI even when the Java Api client has problems or a new but necessary feature on the client side is still in development.

It shall also be possible to test and develop the WebUI application without any running SecHub server or PDS.

### Java API changes
- We extract the methods of `SecHubClient` to an interface and the existing client implementation will be called `DefaultSecHubClient`.
- We separate parts which will be same in mock variant into an abstract base class `AbstractSecHubClient`
- We introduce `MockedSecHubClient` which uses internally only some predefined data and does not really communicate with the SecHub server. Logic shall work, but only in a reduced way that UI development is possible. The mocked client shall have methods to alter its behavior.

### Afterwards
Inside the WebUI a special spring boot profile can be introduced which uses either the default sechub client or a mocked variant - but this shall be done inside another issue.
